packages/gateway-trpc/src/modules/company/company-router.ts

```diff
@@ -1,16 +1,27 @@
 import { z } from "zod"
-import { t } from "../../trpc"
+import { protectedProcedure, t } from "../../trpc"
 import { CompanyWriteSchema } from "@dotkomonline/types"
 import { PaginateInputSchema } from "@dotkomonline/core"
+import { companyEventRouter } from "./company-event-router"
 
 export const companyRouter = t.router({
   create: t.procedure.input(CompanyWriteSchema).mutation(({ input, ctx }) => {
     return ctx.companyService.createCompany(input)
   }),
+  edit: protectedProcedure
+    .input(
+      CompanyWriteSchema.required({
+        id: true,
+      })
+    )
```

Comment:
I prefer to write it as two parameters to keep the WriteSchema clean

```suggestion
      z.object({
        id: CompanySchema.shape.id,
        input: CompanyWriteSchema
      })
    )
```

Reply:
Thats neater yeah